lib/evm/eei.ts

```diff
@@ -522,11 +540,6 @@ export default class EEI {
         // push the created address to the stack
         return new BN(results.createdAddress)
       }
-    } else {
-      // creation failed so don't increment the nonce
-      if (results.vm.createdAddress) {
```

Comment:
`results.vm.createdAddress` is always undefined, as `createdAddress` is set on `results` itself. I fixed this, which led to some test failing. Had a look at [geth](https://github.com/ethereum/go-ethereum/blob/master/core/vm/evm.go#L372) and the contract creator's nonce is incremented, but not decremented on failure.

Reply:
:+1: